### PR TITLE
Second Update to GMB Stats Pane

### PR DESCRIPTION
**Pull request by cbauerman**
## Specs

- [x] Misc style changes
- [x] Move state from `GoogleMyBusiness` to `ui`
- [x] Address a11y violation with select  `onChange` handler 
- [x] Refactor `chart` sub component with React 16.3 lifecycle functions
- [x] Style Pie Charts with Legends to the right side
- [x] Add graceful handling of empty data set

image

### PieChart

1. Navigate to the `PieChart` example page, http://calypso.localhost:3000/devdocs/design/pie-chart locally
2. Press "Show Data Controls" in the header, and use them to verify the chart and legend of the pie chart re-render appropriately as the data changes
screen shot 2018-04-25 at 10 45 02 am
 
### Google My Business Stats Page

1. Connect a site to Google My Business
2. Navigate to the Google My Business Stats page
3. Verify that as the stats are loading a placeholder is shown for the charts
example_loading
4. Change the interval on each chart and verify a placeholder is shown as the data loads 

## Review

- [ ] Product
- [ ] Code

### CODE REVIEW

File: client/state/ui/google-my-business/selector.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ */
+import { get } from 'lodash';
+
+/**
+ * Returns the current interval of a Google My Business stats chart or the
+ * default value, 'week'
+ *
+ * @param  {Object} state Global state tree
+ * @param  {Number} siteId Site ID
+ * @param  {String} statType 'QUERIES' | 'VIEWS' | 'ACTIONS'
```
Comment (yurynix) [inline client/state/ui/google-my-business/selector.js]: I've lower-cased that in my PR

**Comment by yurynix** [comment]
I've taken it for a spin, and for some reason when I choose 'quarter' or 'month',
the pie itself gets a bit larger...

graph

File: client/my-sites/google-my-business/stats/chart.js [hunk 2]
```diff
@@ -91,7 +86,7 @@ class GoogleMyBusinessStatsChart extends Component {
 							<hr className="gmb-stats__metric-hr" />
 						</div>
 					) }
-					<select value={ interval } onChange={ this.onIntervalChange }>
+					<select value={ interval } onChange={ this.changeInterval }>
 						<option value="week">{ 'Week' }</option>
 						<option value="month">{ 'Month' }</option>
 						<option value="quarter">{ 'Quarter' }</option>
```
Comment (stephanethomas) [inline client/my-sites/google-my-business/stats/chart.js]: We should localize those options.

File: client/my-sites/google-my-business/stats/chart.js [hunk 3]
```diff
@@ -118,41 +120,52 @@ class GoogleMyBusinessStatsChart extends Component {
...
-	renderChart() {
-		const { chartTitle, chartType } = this.props;
+	renderPieChart() {
+		const { chartTitle, dataSeriesInfo } = this.props;
 		const { transformedData } = this.state;
 
-		if ( ! transformedData ) {
-			return null;
-		}
+		return !! transformedData ? (
```
Comment (Tug) [inline client/my-sites/google-my-business/stats/chart.js]: Using this construct would be clearer imo:
```
if ( ! transformedData ) {
    // return placeholder
}

// return default
```

File: client/my-sites/google-my-business/stats/style.scss [hunk 4]
```diff
@@ -60,3 +61,14 @@
 		width: 100%;
 	}
 }
+
+// .gmb-stats__metric-chart {
```
Comment (Tug) [inline client/my-sites/google-my-business/stats/style.scss]: This should be removed I guess

**Comment by Tug** [comment]
Code looks good, I had a few minor comments that I think we should address before merging

File: client/my-sites/google-my-business/stats/chart.js [hunk 5]
```diff
@@ -118,41 +120,59 @@ class GoogleMyBusinessStatsChart extends Component {
 		);
 	}
 
-	onIntervalChange = event =>
+	changeInterval = event =>
```
Comment (Tug) [inline client/my-sites/google-my-business/stats/chart.js]: we usually prefix those handlers with `handle` to make it clear that it's a handler:
```
handleChangeInterval = event => ...
``
`

**Comment by Tug** [comment]
LGTM 👍

**Pull request merged**